pkg/health/client/client.go

```diff
@@ -154,7 +209,8 @@ func PathIsHealthy(cp *models.PathStatus) bool {
 		cp.HTTP,
 	}
 	for _, status := range statuses {
-		if !connectivityStatusHealthy(status) {
+		switch GetConnectivityStatusType(status) {
+		case ConnStatusUnreachable:
```

Comment:
nit: I find it a bit weird that a function named `PathIsHealthy` now returns `true` if the state is actually unknown. The new logic and implementation itself is fine, just the old name does not match its new purpose.

I would prefer it if we renamed it to `pathIsUnreachable` or something (of course also invert the output). We can also make it private, as it seems unused outside of the package now.

Similarly, we could also rename `nodeIsHealthy` to `nodeIsUnreachable`.

Reply:
Yes, the name of that function might be misleading. I will change it

I also don't like the fact that it's not very clear what connectivity status do we have from the API model. We have the `Status` string value, we used to assume the non-healthy status if the string is non-empty. I find it quite misleading too. At the beginning I was thinking if I should start fixing that bug with changing the API structure, but then I decided not to do it, because it might be too intrusive for backporting.

But maybe changing the API would be still good as a separate refactoring PR just for 1.9.

Reply:
I will change the function name to `pathIsHealthyOrUnknown` just to avoid confusing double negations like `if !pathIsUnreachable(cp)`. Is it OK for you?

Reply:
`pathIsHealthyOrUnknown` sounds good to me! :+1: 

> But maybe changing the API would be still good as a separate refactoring PR just for 1.9.

Agreed!